### PR TITLE
[Merged by Bors] - feat: lemmas about `Bool.toNat`

### PR DESCRIPTION
**Pull request by alexkeizer**
---

Extracted from #5920 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Bool/Basic.lean [hunk 1]
```diff
@@ -307,11 +307,23 @@ def toNat (b : Bool) : Nat :=
   cond b 1 0
 #align bool.to_nat Bool.toNat
 
+lemma toNat_le_one (b: Bool) : b.toNat ≤ 1 := by
```
Comment (eric-wieser) [inline Mathlib/Data/Bool/Basic.lean:310]:
```suggestion
lemma toNat_le_one (b : Bool) : b.toNat ≤ 1 := by
```

File: Mathlib/Data/Bool/Basic.lean [hunk 2]
```diff
@@ -307,11 +307,23 @@ def toNat (b : Bool) : Nat :=
...
 #align bool.of_nat Bool.ofNat
 
+@[simp] lemma toNat_true  : toNat true = 1  := rfl
+@[simp] lemma toNat_false : toNat false = 0 := rfl
+
+@[simp] lemma toNat_beq_zero (b) : (Bool.toNat b == 0) = !b := by
+  cases b <;> rfl
+
+@[simp] lemma toNat_bne_zero (b) : (Bool.toNat b != 0) = b := by
+  rw [bne, toNat_beq_zero, not_not]
```
Comment (eric-wieser) [inline Mathlib/Data/Bool/Basic.lean:325]: Presumably you want versions of these for `== 1` too?

File: Mathlib/Data/Bool/Basic.lean [hunk 3]
```diff
@@ -307,11 +307,23 @@ def toNat (b : Bool) : Nat :=
...
+
 /-- convert a `ℕ` to a `Bool`, `0 -> false`, everything else -> `true` -/
 def ofNat (n : Nat) : Bool :=
   decide (n ≠ 0)
 #align bool.of_nat Bool.ofNat
 
+@[simp] lemma toNat_true  : toNat true = 1  := rfl
+@[simp] lemma toNat_false : toNat false = 0 := rfl
+
+@[simp] lemma toNat_beq_zero (b) : (Bool.toNat b == 0) = !b := by
```
Comment (eric-wieser) [inline Mathlib/Data/Bool/Basic.lean:321]:
```suggestion
@[simp] lemma toNat_beq_zero (b : Bool) : (b.toNat == 0) = !b := by
``
and below

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

File: Mathlib/Data/Bool/Basic.lean (hunk 2 above)
Comment (alexkeizer) [reply Mathlib/Data/Bool/Basic.lean]: Sure, I've added those and made the formatting a bit more compact

**Pull request closed without merging**